### DISCUSSION TITLE
Adding static assets

### DISCUSSION THREAD

**Discussion by dbersan** [carla-simulator/carla#4330] [Q&A]
Is there any tutorial explaining how to add new static assets to carla? 

More speficially, add new assets to the [Carla Blueprint Library](https://carla.readthedocs.io/en/latest/bp_library/), such that they could be placed on the map via code (eg. `static.prop.guitarcase`, `static.prop.garbage04`, `static.prop.plantpot01` ...)

To be more clear, I want to create new static assets that can be placed on the map. I don't want to know how to place a `static.prop.guitarcase` on the map. I am asking how could I create something similar to a `static.prop.guitarcase`.

**Comment by corkyw10** [comment]
Hello Dhiego, we don't have any official tutorials like that, but we can look into making one.

**Reply by dbersan** [reply]
Thank you. I found a tutorial to adding a prop [here](https://carla.readthedocs.io/en/latest/tuto_A_add_props/), but I guess it isn't the same thing, yeah? Because as I understand, a prop isn't an actor, so it would not be 'spawnable' 

How would the process to add new actors be like? Would it be similar to adding a **car**, or adding a **prop**?

**Reply by dbersan** [reply]
One other question, is it possible to add such "actor assets" to a carla package, that one may import to a Carla install using the `./ImportAssets.sh` script?

**Reply by corkyw10** [reply]
I think I misunderstood your original question, I thought you were looking for a tutorial more on how to build a prop in a 3D editor. In that case, yes props are added to the editor using the instructions in the tutorial you shared. They will be created inside a package folder and then you will be able to make an [importable package](https://carla.readthedocs.io/en/latest/tuto_A_create_standalone/) from them by running the command 
`make package ARGS="packages=<package_name>"`

Props are spawnable like other actors, you can spawn any of the objects in the blueprint library.

**Reply by dbersan** [reply]
Ok, thank you!

So lets say now that I want to add a new **car**, instead of a prop. This [tutorial](https://youtu.be/mHiUUZ4xC9o?t=440) shows how one could create a new "Assets Package" (that is importable using `./ImportAssets.sh`) for a **map**, using the carla docker method. He added a `.fbx` file to a folder, which contains the geometry of the map, and also a `.xodr` file for traffic configurations, and then called the `docker_tools.py`, which outputs the importable package. For adding a **prop**, as I understand, it would be a very similar process (but without needing a `.xodr` file, etc). 

So, how would it be done for a car? Would I just follow this tutorial [here](https://carla.readthedocs.io/en/latest/tuto_A_add_vehicle/), and at the end I put the vehicle folder (originally created at `Content/Carla/Blueprints/Vehicles/<vehicle_name>`) as an argument for the input folder for `docker_tools.py`, that's it? (Or, using the Source method, put this `<vehicle_name>` folder inside `Import/` folder and call `make import`)

**Comment by ronyshaji** [comment]
@dhiegomaga Hey, did you find a method to add the assets ? I am now stuck with this and can't find a method..
